NPE in "group by" combined with "for at"

In versions 2.2 and 3.0.RC1, everything worked as expected. After migrating onto the last build I got this error (when trying to access a particular protected site, other sites work well). I guess it has something to do with the rewriting:

```
HTTP ERROR 500

Problem accessing /exist/apps/test-app/create-ejournals-list.html. Reason:

    Server Error

Caused by:

javax.servlet.ServletException: An unknown error occurred: null
    at org.exist.http.servlets.EXistServlet.doPost(EXistServlet.java:460)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:755)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
    at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:684)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:503)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)
    at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:575)
    at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)
    at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)
    at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
    at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:229)
    at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:103)
    at org.exist.http.urlrewrite.Forward.doRewrite(Forward.java:50)
    at org.exist.http.urlrewrite.XQueryURLRewrite.doRewrite(XQueryURLRewrite.java:549)
    at org.exist.http.urlrewrite.XQueryURLRewrite.applyViews(XQueryURLRewrite.java:414)
    at org.exist.http.urlrewrite.XQueryURLRewrite.service(XQueryURLRewrite.java:360)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
    at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:684)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1496)
    at de.betterform.agent.web.filter.XFormsFilter.doFilter(XFormsFilter.java:171)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1476)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:501)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)
    at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:533)
    at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)
    at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)
    at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
    at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:154)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)
    at org.eclipse.jetty.server.Server.handle(Server.java:370)
    at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:494)
    at org.eclipse.jetty.server.AbstractHttpConnection.content(AbstractHttpConnection.java:982)
    at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.content(AbstractHttpConnection.java:1043)
    at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:865)
    at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:240)
    at org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:82)
    at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:696)
    at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:53)
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:608)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:543)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
    at org.exist.xquery.GroupByClause$GroupByData.access$200(GroupByClause.java:28)
    at org.exist.xquery.GroupByClause.eval(GroupByClause.java:79)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.DynamicTypeCheck.eval(DynamicTypeCheck.java:61)
    at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:323)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:227)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.value.FunctionReference.eval(FunctionReference.java:81)
    at org.exist.xquery.DynamicFunctionCall.eval(DynamicFunctionCall.java:76)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.SwitchExpression.eval(SwitchExpression.java:104)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:323)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:227)
    at org.exist.xquery.DynamicCardinalityCheck.eval(DynamicCardinalityCheck.java:74)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:209)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:323)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:227)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:100)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:100)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.EnclosedExpr.eval(EnclosedExpr.java:81)
    at org.exist.xquery.ElementConstructor.eval(ElementConstructor.java:318)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.EnclosedExpr.eval(EnclosedExpr.java:81)
    at org.exist.xquery.ElementConstructor.eval(ElementConstructor.java:318)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.EnclosedExpr.eval(EnclosedExpr.java:81)
    at org.exist.xquery.ElementConstructor.eval(ElementConstructor.java:318)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.EnclosedExpr.eval(EnclosedExpr.java:81)
    at org.exist.xquery.ElementConstructor.eval(ElementConstructor.java:318)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.EnclosedExpr.eval(EnclosedExpr.java:81)
    at org.exist.xquery.ElementConstructor.eval(ElementConstructor.java:318)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.EnclosedExpr.eval(EnclosedExpr.java:81)
    at org.exist.xquery.ElementConstructor.eval(ElementConstructor.java:318)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.isEmpty(DeferredFunctionCall.java:198)
    at org.exist.xquery.DynamicCardinalityCheck.eval(DynamicCardinalityCheck.java:76)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:209)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:323)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:227)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:100)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:323)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:227)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:323)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:227)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.XQuery.execute(XQuery.java:270)
    at org.exist.xquery.XQuery.execute(XQuery.java:190)
    at org.exist.http.RESTServer.executeXQuery(RESTServer.java:1497)
    at org.exist.http.RESTServer.doPost(RESTServer.java:672)
    at org.exist.http.servlets.EXistServlet.doPost(EXistServlet.java:433)
    ... 46 more

Caused by:

java.lang.NullPointerException
    at org.exist.xquery.GroupByClause$GroupByData.access$200(GroupByClause.java:28)
    at org.exist.xquery.GroupByClause.eval(GroupByClause.java:79)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.DynamicTypeCheck.eval(DynamicTypeCheck.java:61)
    at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:323)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:227)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.value.FunctionReference.eval(FunctionReference.java:81)
    at org.exist.xquery.DynamicFunctionCall.eval(DynamicFunctionCall.java:76)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.SwitchExpression.eval(SwitchExpression.java:104)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:323)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:227)
    at org.exist.xquery.DynamicCardinalityCheck.eval(DynamicCardinalityCheck.java:74)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:209)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:323)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:227)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:100)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:100)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.EnclosedExpr.eval(EnclosedExpr.java:81)
    at org.exist.xquery.ElementConstructor.eval(ElementConstructor.java:318)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.EnclosedExpr.eval(EnclosedExpr.java:81)
    at org.exist.xquery.ElementConstructor.eval(ElementConstructor.java:318)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.EnclosedExpr.eval(EnclosedExpr.java:81)
    at org.exist.xquery.ElementConstructor.eval(ElementConstructor.java:318)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.EnclosedExpr.eval(EnclosedExpr.java:81)
    at org.exist.xquery.ElementConstructor.eval(ElementConstructor.java:318)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.EnclosedExpr.eval(EnclosedExpr.java:81)
    at org.exist.xquery.ElementConstructor.eval(ElementConstructor.java:318)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.SequenceConstructor.eval(SequenceConstructor.java:84)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.EnclosedExpr.eval(EnclosedExpr.java:81)
    at org.exist.xquery.ElementConstructor.eval(ElementConstructor.java:318)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:102)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.isEmpty(DeferredFunctionCall.java:198)
    at org.exist.xquery.DynamicCardinalityCheck.eval(DynamicCardinalityCheck.java:76)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:209)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:323)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:227)
    at org.exist.xquery.ConditionalExpression.eval(ConditionalExpression.java:100)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall$DeferredFunctionCallImpl.execute(FunctionCall.java:420)
    at org.exist.xquery.DeferredFunctionCall.realize(DeferredFunctionCall.java:57)
    at org.exist.xquery.DeferredFunctionCall.iterate(DeferredFunctionCall.java:230)
    at org.exist.xquery.value.ValueSequence.addAll(ValueSequence.java:167)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.TypeswitchExpression.eval(TypeswitchExpression.java:103)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:323)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:227)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:65)
    at org.exist.xquery.ForExpr.processItem(ForExpr.java:239)
    at org.exist.xquery.ForExpr.eval(ForExpr.java:179)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.UserDefinedFunction.eval(UserDefinedFunction.java:155)
    at org.exist.xquery.FunctionCall.evalFunction(FunctionCall.java:323)
    at org.exist.xquery.FunctionCall.eval(FunctionCall.java:227)
    at org.exist.xquery.DebuggableExpression.eval(DebuggableExpression.java:58)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.LetExpr.eval(LetExpr.java:111)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.PathExpr.eval(PathExpr.java:267)
    at org.exist.xquery.AbstractExpression.eval(AbstractExpression.java:71)
    at org.exist.xquery.XQuery.execute(XQuery.java:270)
    at org.exist.xquery.XQuery.execute(XQuery.java:190)
    at org.exist.http.RESTServer.executeXQuery(RESTServer.java:1497)
    at org.exist.http.RESTServer.doPost(RESTServer.java:672)
    at org.exist.http.servlets.EXistServlet.doPost(EXistServlet.java:433)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:755)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
    at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:684)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:503)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)
    at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:575)
    at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)
    at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)
    at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
    at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:229)
    at org.eclipse.jetty.server.Dispatcher.forward(Dispatcher.java:103)
    at org.exist.http.urlrewrite.Forward.doRewrite(Forward.java:50)
    at org.exist.http.urlrewrite.XQueryURLRewrite.doRewrite(XQueryURLRewrite.java:549)
    at org.exist.http.urlrewrite.XQueryURLRewrite.applyViews(XQueryURLRewrite.java:414)
    at org.exist.http.urlrewrite.XQueryURLRewrite.service(XQueryURLRewrite.java:360)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
    at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:684)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1496)
    at de.betterform.agent.web.filter.XFormsFilter.doFilter(XFormsFilter.java:171)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1476)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:501)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)
    at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:533)
    at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)
    at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)
    at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
    at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:154)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)
    at org.eclipse.jetty.server.Server.handle(Server.java:370)
    at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:494)
    at org.eclipse.jetty.server.AbstractHttpConnection.content(AbstractHttpConnection.java:982)
    at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.content(AbstractHttpConnection.java:1043)
    at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:865)
    at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:240)
    at org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:82)
    at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:696)
    at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:53)
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:608)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:543)
    at java.lang.Thread.run(Thread.java:745)

```

```
Caused by: java.lang.NullPointerException
    at org.exist.xquery.GroupByClause$GroupByData.access$200(GroupByClause.java:28)
    at org.exist.xquery.GroupByClause.eval(GroupByClause.java:79)
```

there have been fixes in RC2 (https://github.com/eXist-db/exist/pulls?utf8=✓&q=is%3Apr+is%3Aclosed+group+by) so it might actually be solved right now.

To judge about the (a) bug, we need to have a runnable (piece of) xquery code and some date.....

--- a/src/org/exist/xquery/ForExpr.java
+++ b/src/org/exist/xquery/ForExpr.java
@@ -152,8 +152,14 @@ public class ForExpr extends BindingExpression {
             // calling the where expression for each item in the input sequence)
             // This is possible if the input sequence is a node set and has no
             // dependencies on the current context item.
-            if (at == null && returnExpr instanceof FLWORClause && isOuterFor) {
-                in = ((FLWORClause)returnExpr).preEval(in);
+            if (isOuterFor) {
+                if (returnExpr instanceof WhereClause) {
+                    if (at == null) {
+                        in = ((WhereClause) returnExpr).preEval(in);
+                    }
+                } else if (returnExpr instanceof FLWORClause) {
+                    in = ((FLWORClause) returnExpr).preEval(in);
+                }
             }
 
             final IntegerValue atVal = new IntegerValue(1);

--- a/test/src/xquery/xquery3/groupby.xql
+++ b/test/src/xquery/xquery3/groupby.xql
@@ -599,3 +599,12 @@ declare
 function groupby:recursive-groupby() {
     groupby:recursive-groupby($groupby:values-recursive/v)
 };
+
+declare
+%test:assertEquals(1, 2)
+function groupby:issue-967() {
+    for $nr at $pos in (1,2)
+    group by $pos
+    return
+    $nr
+};

